### PR TITLE
Added high entropy user-agent client hints context option

### PR DESCRIPTION
**Pull request by liljenback**
## Description

Added an optional setting for context properties to collect high entropy user-agent client hints `highEntropyUserAgentHints`.

## Related Issue

[PDCL-8422](https://jira.corp.adobe.com/browse/PDCL-8422)

## Motivation and Context

Regular user-agent is being reduced in Chromium browsers where some information is only made available through this new API.

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.
- [x] I've updated the schema in extension.json or no changes are necessary.
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Review by carterworks — APPROVED** [review]

File: src/view/configuration/dataCollectionSection.jsx [hunk 1]
```diff
@@ -62,7 +72,12 @@ export const bridge = {
     downloadLinkQualifier:
       "\\.(exe|zip|wav|mp3|mov|mpg|avi|wmv|pdf|doc|docx|xls|xlsx|ppt|pptx)$",
     contextGranularity: CONTEXT_GRANULARITY.ALL,
-    context: contextOptions.map(option => option.value)
+    context: contextOptions.reduce((result, option) => {
```
[the remark this answers is not in the document]
Comment (jfkhoury) [reply src/view/configuration/dataCollectionSection.jsx:75]: I agree, might be more readable

File: src/view/dataElements/xdmObject/constants/defaultFormState.js [hunk 2]
```diff
@@ -66,5 +71,11 @@ export default {
   "web.webPageDetails": CONTEXT_WEB,
   "web.webPageDetails.URL": CONTEXT_WEB,
   "web.webReferrer": CONTEXT_WEB,
-  "web.webReferrer.URL": CONTEXT_WEB
+  "web.webReferrer.URL": CONTEXT_WEB,
+  highEntropyUserAgentHints: CONTEXT_HIGH_ENTROPY_USER_AGENT_HINTS,
```
Comment (jonsnyder) [inline src/view/dataElements/xdmObject/constants/defaultFormState.js:75]: We discussed on Slack how this needs to change.

File: test/functional/configuration/configuration.spec.js [hunk 3]
```diff
@@ -140,6 +141,7 @@ test("initializes form fields with full settings", async () => {
   await instances[1].specificContext.deviceField.expectUnchecked();
   await instances[1].specificContext.environmentField.expectUnchecked();
   await instances[1].specificContext.placeContextField.expectUnchecked();
+  await instances[1].specificContext.highEntropyUserAgentHintsContextField.expectUnchecked();
 
   await instancesTabs.selectTab("alloy3");
```
Comment (jonsnyder) [inline test/functional/configuration/configuration.spec.js:147]: Add or augment test cases to make sure that:
 * checking the new checkbox leads to the correct context array in settings.
 * loading the settings with the new context automatically checks the new checkbox.

**Review by jonsnyder — CHANGES_REQUESTED** [review]
Thanks for tackling this! Sorry to throw you into the deep end with the changes necessary to the XDM object data element.

**Review by ninaceban — APPROVED** [review]
looks good to me!

**Review by jonsnyder — APPROVED** [review]

**Pull request merged**